--- models.old.py	2014-11-18 12:12:12.289169335 +0100
+++ models.py	2014-11-18 12:11:31.157171030 +0100
@@ -118,7 +118,7 @@
                             **kwargs)
                     
                     def maybe_alias(table):
-                        if self.query.table_map.has_key(table):
+                        if table in self.query.table_map:
                             return self.query.table_map[table][0]
                         return table
                     
@@ -153,11 +153,7 @@
                                 name = field[0]
                             else:
                                 name = field
-                            if self.query.model._meta.get_field_by_name(name)[
-                                0].db_type().startswith('varchar'):
-                                return 'CAST (T."%s" AS TEXT)' % name
-                            else:
-                                return 'T."%s"' % name
+                            return 'T."%s"' % name
 
                     # The primary key is used in the path; in case it is of a
                     # custom type, ensure appropriate casting is performed. This
